community/en-us/development/submit-code.md

```diff
@@ -38,11 +38,19 @@ git checkout -b dev-1.0 upstream/dev-1.0
 git push --set-upstream origin dev-1.0
 ```
 
-* After modifying the code locally, submit it to your own repository:
+* Create new branch
+```
+git checkout -b xxx origin/dev
```

Comment:
Why do we need this? I think we already have `checkout -b` command in line 37, which is `git checkout -b dev-1.0 upstream/dev-1.0`

Reply:
checking out  from official's repository will can not push user's repository,if coder can not understand Git totally  and execute "git checkout -b dev-1.0 upstream/dev-1.0" the command.
![image](https://user-images.githubusercontent.com/19434382/156093308-6cb44234-d82c-4edc-b696-ecab8693e4ab.png)
if pushing commit in the idea Git GUI.it happens as below.
![image](https://user-images.githubusercontent.com/19434382/156093441-db59637a-ae2b-4be7-8845-4ce7b2e1a7ad.png)
well ,it can not push directly the commit.because upstream repository is official repository.
When pushing to the official repository,it pops up on this window.Next,the coder will login via github.
Even if coder login via github,he still  doesn't push the commit.because the officical repository is not authorized to him.
![image](https://user-images.githubusercontent.com/19434382/156093797-a4f3868a-dac7-4aca-a5da-ce2b06ba69d3.png)
the quick solution is the next picture.
![image](https://user-images.githubusercontent.com/19434382/156094559-85bb98d9-5fa1-47d7-97f7-3ff87b3cc873.png)
But,most of coder can not easily find the solution ,and they think it is self error or network problem so that wasting  a lot of time .  

If  executing "git checkout -b xxx origin/dev"  the command,it is convienient and effecitive to push user's repository directly in the Git GUI.
![image](https://user-images.githubusercontent.com/19434382/156095113-82cff2fc-5a89-40f7-bb5b-a01397e0ccae.png)

Conclusion,"git checkout -b dev-1.0 upstream/dev-1.0' the command is ok,But pushing commit is complicated and misunderstanding for the first contributor because git thinks the commit pushes the upstream repository.

It is my thought.

Reply:
"git checkout -b dev-1.0 origin/dev' the command is friendly for the first contributor because git thinks the commit pushes the user's repository.

Reply:
Make sence